src/lightning_app/components/serve/auto_scaler.py

```diff
@@ -578,16 +580,20 @@ def scale(self, replicas: int, metrics: dict) -> int:
             The target number of running works. The value will be adjusted after this method runs
             so that it satisfies ``min_replicas<=replicas<=max_replicas``.
         """
-        pending_requests_per_running_or_pending_work = metrics["pending_requests"] / (
-            replicas + metrics["pending_works"]
-        )
+        pending_requests = metrics["pending_requests"]
+        active_or_pending_works = replicas + metrics["pending_works"]
+
+        if active_or_pending_works == 0:
+            return 1 if pending_requests > 0 else 0
```

Comment:
Should we use a timeout there ? This might trigger abusive scale down.

Reply:
This the default `scale` function that can be overridden by the user in the `AutoScale` component for controlling scale out and in

Reply:
the scale-in and out timeouts are also controlled by the user as argument to `AutoScaler` component.

Reply:
Done in this PR: https://github.com/Lightning-AI/lightning/pull/16093/files